x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/xcontent/XContentUtils.java

```diff
@@ -102,26 +105,40 @@ public static void addAuthorizationInfo(final XContentBuilder builder, final Map
             return;
         }
         builder.startObject("authorization");
-        switch (authenticationSubject.getType()) {
-            case USER -> builder.array(User.Fields.ROLES.getPreferredName(), authenticationSubject.getUser().roles());
+        addSubjectInfo(builder, authenticationSubject);
+        builder.endObject();
+    }
+
+    private static void addSubjectInfo(XContentBuilder builder, Subject subject) throws IOException {
+        switch (subject.getType()) {
+            case USER -> builder.array(User.Fields.ROLES.getPreferredName(), subject.getUser().roles());
             case API_KEY -> {
-                builder.startObject("api_key");
-                Map<String, Object> metadata = authenticationSubject.getMetadata();
-                builder.field("id", metadata.get(AuthenticationField.API_KEY_ID_KEY));
-                Object name = metadata.get(AuthenticationField.API_KEY_NAME_KEY);
-                if (name instanceof String) {
-                    builder.field("name", name);
-                }
-                builder.endObject();
+                addApiKeyInfo(builder, subject);
             }
-            case SERVICE_ACCOUNT -> builder.field("service_account", authenticationSubject.getUser().principal());
+            case SERVICE_ACCOUNT -> builder.field("service_account", subject.getUser().principal());
             case CROSS_CLUSTER_ACCESS -> {
```

Comment:
Based on the callers of this method, a cross-cluster authentication should never reach here. Therefore it can be argued to not handle it at all, e.g. just return without doing anything. However since the method itself is not prescribed to be used only by existing callers, it feels more appropriate to handle the case for self-consistency.

Reply:
++ on just supporting it